### PR TITLE
Updates navigation response

### PR DESCRIPTION
**Pull request by d1am0nd**

### CODE REVIEW

File: src/types/navigation.ts [hunk 1]
```diff
@@ -17,9 +17,28 @@ export type NavigationItemExternal = NavigationItem & {
   };
 };
 
+export type NavigationItemExtraFilter = {
+  include?: number[];
+  exclude?: number[];
```
Comment (HenriBeck) [inline src/types/navigation.ts]: I don't think exclude makes any sense, or would you disagree?

File: src/types/navigation.ts [hunk 2]
```diff
@@ -17,9 +17,28 @@ export type NavigationItemExternal = NavigationItem & {
...
+
 export type NavigationItemCategory = NavigationItem & {
   type: 'category';
-  extraFilters: {[key: string]: {include: number[]}}[];
+  extraFilters: {
+    [key: string]:
+      | NavigationItemExtraFilter
+      | NavigationItemAttributeExtraFilter[];
+  }[];
   categoryId: number;
```
Comment (HenriBeck) [inline src/types/navigation.ts:42]: @d1am0nd let's also add the `category` property here

File: src/types/navigation.ts [hunk 3]
```diff
@@ -17,9 +17,28 @@ export type NavigationItemExternal = NavigationItem & {
...
+  };
+};
+
 export type NavigationItemCategory = NavigationItem & {
   type: 'category';
-  extraFilters: {[key: string]: {include: number[]}}[];
+  extraFilters: {
+    [key: string]:
+      | NavigationItemExtraFilter
+      | NavigationItemAttributeExtraFilter[];
```
Comment (HenriBeck) [inline src/types/navigation.ts:40]: I don't think this type is correct if I look at the latest structure:

```
"extraFilters": {
              "merchants": {
                "include": [
                  1
                ]
              },
              "attributes": [
                {
                  "attribute": {
                    "id": 1105,
                    "key": "heelHeightGroup",
                    "label": "Absatzhöhengruppe",
                    "type": "",
                    "multiSelect": false
                  },
                  "include": [
                    1454,
                    1463,
                    1478,
                    1427
                  ]
                }
              ],
              "product_ids": {
                "include": [
                  "41",
                  "38",
                  "12"
                ]
              }
            },
```

Comment (d1am0nd) [reply src/types/navigation.ts:40]: It looks fine to me 🤔 

I also pinged Chintu if there's a better solution and the answer was basically no

**Review by HenriBeck — APPROVED** [review]

**Pull request merged**